08-graphs/02-graphs-algorithms.md

```diff
@@ -130,22 +234,247 @@ In the worst-case you will need to add each node to the Queue, so the space comp
 
 Where breadth-first-search spreads out from a starting node in order of distance from the starting node, depth first search follows each path as far as possible before backing up and following the next closest path.
 
+[HackerEarth](https://www.hackerearth.com/practice/algorithms/graphs/depth-first-search/visualize/) has an excellent description and visualization of the algorithm.
```

Comment:
Similar to BFS, we may consider adding a video here to illustrate how depth first search works when looking at a diagram rather than relying on hacker earth. I would also explicitly call out that because we want to follow each path as far as we can, the algorithm lends itself to using a stack instead of a queue (LIFO v FIFO).